crates/starknet-devnet-server/src/server.rs

```diff
@@ -150,3 +158,19 @@ async fn response_logging_middleware(
     let response = Response::from_parts(parts, body);
     Ok(response)
 }
+
+async fn restrictive_middleware(
+    State(server_config): State<ServerConfig>,
+    request: Request,
+    next: Next,
+) -> Result<impl IntoResponse, (StatusCode, String)> {
+    if let Some(restricted_paths) = &server_config.restricted_methods {
```

Comment:
I think that it can be simplified with a match instead of if

Reply:
how to make it continue with the default behavior if the method is not from the restricted ones

Reply:
Since @mikiw is not likely to respond here, I'll resolve this and the other related comment.